ingestion/src/metadata/profiler/orm/functions/median.py

```diff
@@ -116,10 +116,28 @@ def _(elements, compiler, **kwargs):
 
 @compiles(MedianFn, Dialects.MySQL)
 def _(elements, compiler, **kwargs):  # pylint: disable=unused-argument
-    """Median computation for MySQL currently not supported
-    Needs to be tackled in https://github.com/open-metadata/OpenMetadata/issues/6340
-    """
-    return "NULL"
+    """Median computation for MySQL"""
+    col = compiler.process(elements.clauses.clauses[0])
+    table = elements.clauses.clauses[1].value
+    percentile = elements.clauses.clauses[2].value
+
+    return """
+    (SELECT
+        {col}
+    FROM (
+        SELECT
+            t.{col}, 
+            @row_num := @row_num + 1 AS row_num
```

Comment:
I think we'll need to replace `@row_num := @row_num + 1 AS row_num` with `ROW_NUMBER() OVER () AS row_num`. I get the following warning on my end

```
[Warning: #1287 Setting user variables within expressions is deprecated and will be removed in a future release](https://stackoverflow.com/questions/67256687/mysql-warning-1287-setting-user-variables-within-expressions-is-deprecated-a)
```

Reply:
@TeddyCr what are you using to generate that warning? I'm trying to find the best solution and want to see if my solution is still throwing that warning.  I can use the ROW_NUMBER() replacement, but I needed the final value of `row_num` to calculate row number that matches the percentile.  Thanks.

Reply:
I am using MySQL 8.0.

I am just wondering what is the advantage betweenusing a the variable and a cross join vs the below

```sql
    (
    SELECT
        {col}
     FROM (
         SELECT
             t.{col}, 
             ROW_NUMBER() OVER () AS row_num,
             COUNT(*) OVER() AS counter
         FROM 
            {table} t
         ORDER BY installed_rank
         ) temp
     WHERE
	 	temp.row_num = ROUND({percentile} * counter)
     )
```

Reply:
For testing, I am using the om MySQL docker image with a table of 1 million rows.  The query:
```
    (SELECT
        {col}
    FROM (
        SELECT
            t.{col}, 
            ROW_NUMBER() OVER () AS row_num
        FROM 
            {table} t,
            (SELECT @counter := COUNT(*) FROM {table}) t_count 
        ORDER BY {col}
        ) temp
    WHERE temp.row_num = ROUND({percentile} * @counter)
    )
```
is five to seven times faster with an index and 4 times faster without an index than your query above.

Not knowing the potential impacts on production systems, I tried to find the fastest query I could. Granted my single table is probably an oversimplification.

Reply:
Make sense. That was my assumption, just wanted to confirm. Thanks @schlameel